### PR TITLE
[Feature]Add containerize bento feature for Yatai service

### PR DESCRIPTION
**Pull request by yubozhao**
## Description

#1311 

Add new Python API for building container for BentoService
```python
yatai_client = get_yatai_client()
yatai_client.repository.containerize(bento, tag, repository, build_args, push)
```
**Params:**

* bento (string): Bento id,  NAME:VERSION
* tag (string): desired image tag, if no tag is provided, Yatai will opts for `bento_name:bento_version` tag
* build_args (dict): Additional docker build args
* repository (string): Repository address for pushing the built image.
* push (boolean): Default to False, set True to push the image to repository after build.

CLI Changes:
* Remove `--username` and `--password` option for `bentoml containize` command. It will use the docker auth configuration on the system that runs Yatai Service for pushing to repository.

- [x] Implementation
- [x] Add integration test
- [x] Update API reference on the doc.

## Types of changes

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: tests/integration/yatai_server/test_containerize.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import logging
+
+from bentoml.yatai.client import get_yatai_client
```
Comment (bojiang) [inline tests/integration/yatai_server/test_containerize.py:3]: Is it possible to test a remote yatai service?

File: bentoml/cli/bento_service.py [hunk 2]
```diff
@@ -345,18 +340,15 @@ def install_completion(append, shell, path):
         '--build-arg', multiple=True, help="pass through docker image build arguments"
     )
     @click.option(
-        '-u', '--username', type=click.STRING, required=False,
-    )
-    @click.option(
-        '-p', '--password', type=click.STRING, required=False,
+        '--repository', type=click.STRING, required=False,
```
Comment (parano) [inline bentoml/cli/bento_service.py:343]: what is this parameter? add some docs here?

File: bentoml/cli/bento_service.py [hunk 3]
```diff
@@ -386,73 +378,22 @@ def containerize(bento, push, tag, build_arg, username, password, yatai_url):
...
-                            auth_config=auth_config_payload,
-                        )
-                    ):
-                        _echo(line)
-                _echo(
-                    f'Pushed {tag} to {name}', CLI_COLOR_SUCCESS,
-                )
-            except (docker.errors.APIError, BentoMLException) as error:
-                raise CLIException(f'Could not push Docker image: {error}')
+            _echo(f'Build container image: {tag}', CLI_COLOR_SUCCESS)
```
Comment (parano) [inline bentoml/cli/bento_service.py:396]: check the status code of the return and handle errors...

File: bentoml/cli/bento_service.py [hunk 4]
```diff
@@ -345,18 +340,15 @@ def install_completion(append, shell, path):
...
-        '-u', '--username', type=click.STRING, required=False,
-    )
-    @click.option(
-        '-p', '--password', type=click.STRING, required=False,
+        '--repository', type=click.STRING, required=False,
     )
     @click.option(
         '--yatai-url',
         type=click.STRING,
         help='Remote YataiService URL. Optional. '
```
Comment (parano) [inline bentoml/cli/bento_service.py:348]: `help='Specify the YataiService for running the containerization, default to the Local YataiService with local docker deamon'`

Comment (parano) [reply bentoml/cli/bento_service.py:348]: Could you update the inline docstring below, some of them are outdated

File: bentoml/cli/bento_service.py [hunk 5]
```diff
@@ -386,73 +378,22 @@ def containerize(bento, push, tag, build_arg, username, password, yatai_url):
...
-        _echo(
-            f'Finished building {tag} from {bento}', CLI_COLOR_SUCCESS,
-        )
-
-        if push:
-            auth_config_payload = (
-                {"username": username, "password": password}
-                if username or password
-                else None
+        with Spinner("Building Docker image"):
```
Comment (parano) [inline bentoml/cli/bento_service.py:388]: Change the spinner message to something like this? ```Sending containerize RPC to YataiService {yatai_client.url}``` and for local yatai service use:  ```Containerizing {bento_tag} with local YataiService and docker deamon from env {docker_api.deamon_url}```

Comment (parano) [reply bentoml/cli/bento_service.py:388]: Actually, would it make more sense to move this part to the Python API instead? that way users got to see the message and spinner even when using the Python API from a jupyter notebook

File: bentoml/cli/bento_service.py (hunk 3 above)
Comment (yubozhao) [reply bentoml/cli/bento_service.py:396]: The status code check is handled in the python API. I will double-check to make sure it is there.

File: bentoml/cli/bento_service.py (hunk 5 above)
Comment (yubozhao) [reply bentoml/cli/bento_service.py]: As discussed offline. Spinner is better kept in the CLI client, because of it's implementation.

**Comment by parano** [comment]
The lint and format seems to fail on this PR

**Pull request merged**